crates/oxc_linter/src/rules/jest/no_confusing_set_timeout.rs

```diff
@@ -0,0 +1,322 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{
+    context::LintContext,
+    jest_ast_util::{JestFnKind, JestGeneralFnKind, ParsedGeneralJestFnCall},
+    rule::Rule
+};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(jest/no-confusing-set-timeout)")]
+#[diagnostic(severity(warning))]
+struct NoGlobalSetTimeoutDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(jest/no-confusing-set-timeout)")]
+#[diagnostic(severity(warning))]
+struct NoMultipleSetTimeoutsDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(jest/no-confusing-set-timeout)")]
+#[diagnostic(severity(warning))]
+struct NoUnorderSetTimeoutDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoConfusingSetTimeout;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow confusing usages of jest.setTimeout
+    ///
+    /// ### Why is this bad?
+    ///
+    /// - being called anywhere other than in global scope
+    /// - being called multiple times
+    /// - being called after other Jest functions like hooks, `describe`, `test`, or `it`
+    ///
+    ///
+    /// ### Example
+    ///
+    /// All of these are invalid case:
+    /// ```javascript
+    /// escribe('test foo', () => {
+    ///   jest.setTimeout(1000);
+    ///   it('test-description', () => {
+    ///     // test logic;
+    ///   });
+    /// });
+    ///
+    /// describe('test bar', () => {
+    ///   it('test-description', () => {
+    ///     jest.setTimeout(1000);
+    ///     // test logic;
+    ///   });
+    /// });
+    ///
+    /// test('foo-bar', () => {
+    ///   jest.setTimeout(1000);
+    /// });
+    ///
+    /// describe('unit test', () => {
+    ///   beforeEach(() => {
+    ///     jest.setTimeout(1000);
+    ///   });
+    /// });
+    /// ```
+    NoConfusingSetTimeout,
+    restriction
+);
+
+impl Rule for NoConfusingSetTimeout {
+    fn run_once(&self, ctx: &LintContext) {
+        let mut should_emit_orded_set_timeout = false;
+        let mut seen_jest_timeout = false;
+
+        for node in ctx.nodes().iter() {
```

Comment:
It seems like you are checking for `CallExpression` - you should use the `run` method instead of iterating it here.

Reply:
Yeah. But I thought it's not clean enough before. It needs to record the counts of `jest.setTimeout` call. And if use `run` that the counts should record in the `semantic`.

Reply:
Seems we can get the Identifier `jest` first: `ctx.scopes().root_unresolved_references().iter().filter(|(key, _)| {
           key == 'jest'
        });`

Reply:
Yeah, thanks. It helped me a lot. 

Pray the benchmark can pass. 🙏